File: app/src/main/java/org/mobileProgramming/maintermproject/Fragment2/WaitActivity.java
package org.mobileProgramming.maintermproject.Fragment2;

import androidx.appcompat.app.AppCompatActivity;

import android.app.ActivityOptions;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.database.FirebaseDatabase;

import org.mobileProgramming.maintermproject.Fragment4.MessageActivity;
import org.mobileProgramming.maintermproject.R;
import org.mobileProgramming.maintermproject.model.ReservationModel;

public class WaitActivity extends AppCompatActivity {
    private TextView wait_Salestitle;
    private TextView wait_Address;
    private TextView wait_ReservationTime;
    private Button wait_okay;
    private String myUid = FirebaseAuth.getInstance().getCurrentUser().getUid();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_wait);

        wait_Salestitle = findViewById(R.id.wait_Salestitle);
        wait_Address = findViewById(R.id.wait_address);
        wait_ReservationTime = findViewById(R.id.wait_ReservationTime);
        wait_okay = findViewById(R.id.waitOkay);
        Intent intent = getIntent();

        final String Salestitle = intent.getStringExtra("Title");
        final String address = intent.getStringExtra("address");
        final String ReservationTime = intent.getStringExtra("ReservationTime");
        final String destinationUid = intent.getStringExtra("destinationUid");

        wait_Salestitle.setText(Salestitle);
        wait_Address.setText(address);
        wait_ReservationTime.setText(ReservationTime);

        final Intent messageIntent = new Intent(this, MessageActivity.class);
        messageIntent.putExtra("destinationUid", destinationUid);
        messageIntent.putExtra("Title", Salestitle);
        messageIntent.putExtra("ReservationTime", destinationUid);
        messageIntent.putExtra("address", address);
        wait_okay.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
            ReservationModel reservationModel = new ReservationModel();
            reservationModel.myUiddestinationUid = myUid + destinationUid;
            reservationModel.destinationUidmyUid = destinationUid + myUid;
            reservationModel.Salestitle = Salestitle;
            reservationModel.address = address;
            reservationModel.date = ReservationTime;
            reservationModel.client = myUid;
            reservationModel.manager = destinationUid;

            FirebaseDatabase.getInstance().getReference().child("ReservationInfo")
                 .child(myUid+destinationUid).setValue(reservationModel);

                ActivityOptions activityOptions = null;
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
                    activityOptions = ActivityOptions.makeCustomAnimation(getApplicationContext(), R.anim.fromright, R.anim.toleft);

                    startActivity(messageIntent, activityOptions.toBundle());
                }
            }
        });
    }
}

File: Kotlin/Android_PetCommunity/app/src/main/java/com/sehun/android_pet_community/WaitActivity.kt
package com.sehun.android_pet_community

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.FirebaseFirestore
import com.sehun.android_pet_community.Model.ChatModel
import com.sehun.android_pet_community.Model.ReservationModel
import com.sehun.android_pet_community.databinding.ActivityWaitBinding


class WaitActivity : AppCompatActivity() {
    private  var mbinding : ActivityWaitBinding? = null
    private val binding get() = mbinding!!
    private val myUid = FirebaseAuth.getInstance().currentUser!!.uid

    var firebaseFirestore : FirebaseFirestore? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mbinding = ActivityWaitBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val destinationUid = intent.getStringExtra("destinationUid")
        val ReservationTime = intent.getStringExtra("ReservationTime")
        val address = intent.getStringExtra("address")
        val salestitle = intent.getStringExtra("salestitle")

        binding.waitReservationTime.text = ReservationTime
        binding.waitAddress.text = address
        binding.waitSalestitle.text = salestitle

        val messageIntent = Intent(this, MessageActivity::class.java)
        messageIntent.putExtra("destinationUid", destinationUid)
        messageIntent.putExtra("Title", salestitle)
        messageIntent.putExtra("ReservationTime", destinationUid)
        messageIntent.putExtra("address", address)
        binding.waitOkay.setOnClickListener {
            val reservationModel : ReservationModel = ReservationModel()
            reservationModel.myUiddestinationUid = myUid + destinationUid
            reservationModel.destinationUidmyUid = destinationUid + myUid
            reservationModel.Salestitle = salestitle
            reservationModel.address = address
            reservationModel.date = ReservationTime
            reservationModel.client = myUid
            reservationModel.manager = destinationUid

            FirebaseFirestore.getInstance().collection("ReservationInfo").document(myUid + destinationUid)
                .set(reservationModel)

//            val chatModel = ChatModel()
//            chatModel.users!!.put(myUid, true)
//            chatModel.users!!.put(destinationUid!!, true)
//
//
////            FirebaseFirestore.getInstance().collection("chatrooms").add(chatModel)
//            FirebaseFirestore.getInstance().collection("chatrooms").document("sss").set(chatModel)

            startActivity(messageIntent)
        }
    }

    override fun onDestroy() {
        // onDestroy 에서 binding class 인스턴스 참조를 정리해주어야 한다.
        mbinding = null
        super.onDestroy()
    }

}